File: next/ts/src/core/bootstrapContext.ts
import type { $MMID, $PromiseMaybe } from "../helper/types.ts";
import type { MicroModule } from "./micro-module.ts";
import type { $ConnectResult } from "./nativeConnect.ts";
export interface $BootstrapContext {
  dns: $DnsMicroModule;
}
export interface $DnsMicroModule {
  install(mm: MicroModule): void;
  uninstall(mm: MicroModule): void;
  connect(mmid: $MMID, reason?: Request): $PromiseMaybe<$ConnectResult>;
  query(mmid: $MMID): Promise<MicroModule | undefined>
}

File: android/app/src/main/java/info/bagen/dwebbrowser/microService/core/BootstrapContext.kt
package info.bagen.dwebbrowser.microService.core

import info.bagen.dwebbrowser.microService.helper.Mmid
import org.http4k.core.Request


interface BootstrapContext {
    val dns: DnsMicroModule
}

interface DnsMicroModule {
    /**
     * 动态安装应用
     */
    fun install(mm: MicroModule)

    /**
     * 动态卸载应用
     */
    fun uninstall(mm: MicroModule)

    /**
     * 动态js应用查询
     */
    fun query(mmid: Mmid): MicroModule?

    /**
     * 重启应用
     */
    fun restart(mmid:Mmid)

    /**
     * 与其它应用建立连接
     */
    suspend fun connect(mmid: Mmid, reason: Request? = null): ConnectResult

    /**
     * 启动其它应用
     */
    suspend fun bootstrap(mmid: Mmid)
}